dep-graph/client/src/app/shell.tsx

```diff
@@ -0,0 +1,205 @@
+// nx-ignore-next-line
+import type { DepGraphClientResponse } from '@nrwl/workspace/src/command-line/dep-graph';
+import { useEffect, useState } from 'react';
+import Sidebar from './sidebar/sidebar';
+import { useDepGraph } from './hooks/use-dep-graph';
+import { useEnvironmentConfig } from './hooks/use-environment-config';
+import { useIntervalWhen } from './hooks/use-interval-when';
+import { useProjectGraphDataService } from './hooks/use-project-graph-data-service';
+import Tippy from '@tippyjs/react';
+import { getGraphService } from './machines/graph.service';
+
+export function Shell() {
+  const [state, send] = useDepGraph();
+  const projectGraphService = useProjectGraphDataService();
+  const environment = useEnvironmentConfig();
+
+  const projectSelected = state.context.selectedProjects.length > 0;
+
+  const { lastPerfReport } = state.context;
+
+  const [selectedProjectId, setSelectedProjectId] = useState<string>(
```

Comment:
I'd use `useRef` here since the selected  project ID is only used in other callbacks and does not affect re-render.

The "current" value of `useState` could be stale (mitigated with hook deps), so you could simply do this.

```
const selectedProjectId = useRef(environment.appConfig.defaultProjectGraph);

useEffect(() => {

    const { appConfig } = environment;

    const projectInfo = appConfig.projectGraphs.find(
      (graph) => graph.id === selectedProjectId.current
    );
    // ...
})
```